internal/fleet/agent_policy/schema.go

```diff
@@ -172,6 +173,104 @@ func getSchema() schema.Schema {
 					),
 				},
 			},
+			"advanced_monitoring_options": schema.SingleNestedAttribute{
+				Description: "Advanced monitoring options for the agent policy. Includes HTTP monitoring endpoint configuration and diagnostic settings.",
+				Optional:    true,
+				Computed:    true,
+				Attributes: map[string]schema.Attribute{
+					"http_monitoring_endpoint": schema.SingleNestedAttribute{
+						Description: "HTTP monitoring endpoint configuration for agent health checks and liveness probes.",
+						Optional:    true,
+						Computed:    true,
+						Attributes: map[string]schema.Attribute{
+							"enabled": schema.BoolAttribute{
+								Description: "Enable the HTTP monitoring endpoint. When enabled, exposes a /liveness endpoint for health checks.",
+								Optional:    true,
+								Computed:    true,
+								Default:     booldefault.StaticBool(false),
+							},
+							"host": schema.StringAttribute{
+								Description: "Host for the HTTP monitoring endpoint.",
+								Optional:    true,
+								Computed:    true,
+								Default:     stringdefault.StaticString("localhost"),
+							},
+							"port": schema.Int32Attribute{
+								Description: "Port for the HTTP monitoring endpoint.",
+								Optional:    true,
+								Computed:    true,
+								Default:     int32default.StaticInt32(6791),
+								Validators: []validator.Int32{
+									int32validator.Between(0, 65535),
+								},
+							},
+							"buffer_enabled": schema.BoolAttribute{
+								Description: "Enable monitoring buffer for the HTTP endpoint.",
+								Optional:    true,
+								Computed:    true,
+								Default:     booldefault.StaticBool(false),
+							},
+							"pprof_enabled": schema.BoolAttribute{
+								Description: "Enable /debug/pprof/* profiling endpoints. Warning: enabling this may pose a security risk if the monitoring endpoint is accessible over a network.",
+								Optional:    true,
+								Computed:    true,
+								Default:     booldefault.StaticBool(false),
+							},
+						},
+					},
+					"diagnostics": schema.SingleNestedAttribute{
+						Description: "Diagnostic settings for rate limiting and file upload behavior.",
+						Optional:    true,
+						Computed:    true,
+						Attributes: map[string]schema.Attribute{
+							"rate_limits": schema.SingleNestedAttribute{
+								Description: "Rate limiting configuration for diagnostics requests from Fleet.",
+								Optional:    true,
+								Computed:    true,
+								Attributes: map[string]schema.Attribute{
+									"interval": schema.StringAttribute{
+										Description: "Rate limiting interval for diagnostics requests (e.g., '1m', '30s').",
```

Comment:
I wonder if we should be using a duration here so `60s` and `1m` are equivalent? 

```suggestion
										Description: "Rate limiting interval for diagnostics requests (e.g., '1m', '30s').",
										CustomType: customtypes.DurationType,
```

Reply:
https://github.com/elastic/terraform-provider-elasticstack/pull/1537/commits/9333e3a0cc3ba53c6088c2a95eafaba0b01e82bc